core/app/models/spree/store_credit.rb

```diff
@@ -288,4 +289,11 @@ def associate_credit_type
       self.credit_type = Spree::StoreCreditType.find_by_name(credit_type_name)
     end
   end
+
+  def apply_to_orders_in_confirm_state
```

Comment:
I agree this is a little awkward but it was the best way I could come up with to hook into the creation of store credit and ensure the state machine callbacks are still triggered when transitioning between order states. Open to suggestions of course.